uploadservice/src/main/java/net/gotev/uploadservice/MultipartUploadTask.java

```diff
@@ -143,10 +145,13 @@ private void writeFiles(BodyWriter bodyWriter) throws IOException {
 
     @Override
     protected void onSuccessfulUpload() {
-        for (UploadFile file : params.getFiles()) {
-            addSuccessfullyUploadedFile(file.getPath());
+        // addSuccessfullyUploadedFile removes the file from the list so we have
+        // to use a copy to not get a java.util.ConcurrentModificationException
+        List<UploadFile> files = new ArrayList<>(params.getFiles());
```

Comment:
We can use a list iterator here to avoid java.util.ConcurrentModificationException and to not duplicate the list in memory. I will take care of this after the merge